src/FuncSharp/Strings/NonEmptyString.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+
+namespace FuncSharp;
+
+public sealed class NonEmptyString : IEquatable<string>
+{
+    private NonEmptyString(string value)
+    {
+        Value = value;
+    }
+
+    public string Value { get; }
+
+    public static implicit operator string(NonEmptyString s)
+    {
+        return s.MapRef(v => v.Value);
+    }
+
+    public static NonEmptyString CreateUnsafe(string value)
+    {
+        var result = CreateNullable(value);
+        if (result is null)
+            throw new ArgumentException("You cannot create NonEmptyString from whitespace, empty string or null.");
+        return result;
+    }
+
+    public static IOption<NonEmptyString> Create(string value)
```

Comment:
nit: I still feel as a newcomer when it comes to using this lib so since the other method is called `CreateNullable` so I'd expect this one to be called `CreateOption` or `CreateAsOption`